api/src/main/java/org/pmiops/workbench/monitoring/attachments/AttachmentKey.java

```diff
@@ -0,0 +1,23 @@
+package org.pmiops.workbench.monitoring.attachments;
+
+public enum AttachmentKey implements AttachmentKeyBase {
+  BUFFER_ENTRY_STATUS("buffer_entry_status"),
+  DEBUG_COLOR("debug_color"),
+  NOTEBOOK_NAME("notebook_name"),
+  NOTEBOOK_WORKSPACE_NAMESPACE("notebook_workspace_namespace"),
+  USER_DISABLED("user_disabled"),
+  DATASET_INVALID("dataset_invalid"),
+  WORKSPACE_ACTIVE_STATUS("workspace_active_status"),
+  WORKSPACE_DATA_ACCESS_LEVEL("workspace_data_access_level");
```

Comment:
Not sure how we can make it happen but I think it would be ideal to have a way to associate the metadata keys and the metrics that they are describing. Through code references would be the most ideal, code locality can also work.

Reply:
Interesting. I don't know if anonymous enums are a thing, but if they were, then I could declare them scoped to each curve.

The real problem is that they're many:many w/r/t metrics. Say I had some machines running Linux and others running Solaris or some randomness. Lots of my gauge metrics might want to indicate the OS I was running, and we'd use the same string key for that. OpenCensus doesn't enforce that, but it's part of our model. The attachments refer to the whole MeasureMap (i.e. the MeasurementBundle), so it's key to keep things consistent there.

Some of these I don't want to share (like `user_disabled`), which I think is what you're getting at. We could have a `Set<Metric>` associated with each AttachmentKey if we wanted (or vice-versa). I don't know the rules on enums referring to other enums in their constructors, but I remember from C++ Thou Shalt Not rely on instantiation ordering of static objects.